### PR TITLE
feat(core/inlines): support inline [^element^] linking

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

File: src/core/inlines.js [hunk 1]
```diff
@@ -30,6 +30,18 @@ const inlineVariable = /\B\|\w[\w\s]*(?:\s*:[\w\s&;<>]+)?\|\B/; // |var : Type|
...
 const inlineExpansion = /(?:\[\[\[(?:!|\\|\?)?#?[\w-.]+\]\]\])/; // [[[expand]]]
 const inlineAnchor = /(?:\[=[^=]+=\])/; // Inline [= For/link =]
+const inlineElement = /(?:\[\^(?:!|\\|\?)?[A-Za-z]+\^\])/; // Inline [^element^]
+
+/**
+ * @param {string} matched
+ * @return {HTMLElement}
+ */
+function inlineElementMatches(matched) {
+  const value = matched.slice(2, -2).trim();
```
Comment (sidvishnoi) [inline src/core/inlines.js]: `matched` includes `[[!value]]`, right?

Comment (marcoscaceres) [reply src/core/inlines.js]: oh, it shouldn't be.

File: src/core/inlines.js [hunk 2]
```diff
@@ -30,6 +30,18 @@ const inlineVariable = /\B\|\w[\w\s]*(?:\s*:[\w\s&;<>]+)?\|\B/; // |var : Type|
 const inlineCitation = /(?:\[\[(?:!|\\|\?)?[A-Za-z0-9.-]+\]\])/; // [[citation]]
 const inlineExpansion = /(?:\[\[\[(?:!|\\|\?)?#?[\w-.]+\]\]\])/; // [[[expand]]]
 const inlineAnchor = /(?:\[=[^=]+=\])/; // Inline [= For/link =]
+const inlineElement = /(?:\[\^(?:!|\\|\?)?[A-Za-z]+\^\])/; // Inline [^element^]
```
Comment (sidvishnoi) [inline src/core/inlines.js]:
```suggestion
const inlineElement = /(?:\[\^[A-Za-z]+\^\])/; // Inline [^element^]
```

File: src/core/inlines.js [hunk 3]
```diff
@@ -30,6 +30,18 @@ const inlineVariable = /\B\|\w[\w\s]*(?:\s*:[\w\s&;<>]+)?\|\B/; // |var : Type|
 const inlineCitation = /(?:\[\[(?:!|\\|\?)?[A-Za-z0-9.-]+\]\])/; // [[citation]]
 const inlineExpansion = /(?:\[\[\[(?:!|\\|\?)?#?[\w-.]+\]\]\])/; // [[[expand]]]
 const inlineAnchor = /(?:\[=[^=]+=\])/; // Inline [= For/link =]
+const inlineElement = /(?:\[\^[A-Za-z]+\^\])/; // Inline [^element^]
```
Comment (sidvishnoi) [inline src/core/inlines.js]:
```suggestion
const inlineElement = /(?:\[\^[A-Za-z]+(?:-[A-Za-z]+)?\^\])/; // Inline [^element^]
```
or simpler and the one accepting potentially  bad input:
```suggestion
const inlineElement = /(?:\[\^[A-Za-z-]+\^\])/; // Inline [^element^]
```

**Pull request merged**